server/src/gql/resolver.py

```diff
@@ -72,3 +72,66 @@ def resolve_add_new_mates(
     family.mate_ids += [new_mate.id for new_mate in new_mates]
     family.save()
     return new_mates
+
+
+@mutation.field("assignMatesToFamilies")
+def resolve_assign_mates_to_families(
+    obj: Any, info: GraphQLResolveInfo, newAssignments: List[NewAssignmentInput]
+) -> List[str]:
+    """
+    Assign mates to specified families.
+
+    Args:
+    ----
+        obj: GraphQL
+        info: GraphQLInfo
+        newAssignments: each item of NewAssignmentInput contains
+            - familyId
+            - list of id's of mates to be assigned to that family
+
+    Returns ids of updated mates
+
+    """
+    user = User.lookup_user(current_user.get_id())
+    if not user:
+        raise Exception("User not found")
+
+    moved_mates: List[str] = []
+
+    unassigned_family = user.unassigned_family
+    unassigned_id = str(unassigned_family.id)
+    unassigned_mate_ids = set([str(mate.id) for mate in unassigned_family.mate_ids])
+    family_ids = set([str(family.id) for family in user.family_ids])
+    for input in newAssignments:
+        for mate_id in input["mateIds"]:
+            if mate_id not in unassigned_mate_ids:
```

Comment:
we could probably make this more generic and use it for reassigning mates too - in which case this check doesnt need to be here.